src/script.js

```diff
@@ -1,18 +1,121 @@
   if (keyElement !== undefined) {
     keyElement.dispatchEvent(new Event('mousedown'));
   }
 }
 
-function imitateMouseUp(keyElement) {
+function onSpecialKeyDown(keyContent) {
+  if (keyContent !== undefined) {
+    const keys = [...keyboard.elements.keys];
+    const keyElement = keys.find((element) => element.textContent === keyContent);
+    keyElement.dispatchEvent(new Event('mousedown'));
+  }
+}
+
+function onKeyUp(keyElement) {
   if (keyElement !== undefined) {
     keyElement.dispatchEvent(new Event('mouseup'));
   }
 }
 
+function onSpecialKeyUp(keyContent) {
+  if (keyContent !== undefined) {
+    const keys = [...keyboard.elements.keys];
+    const keyElement = keys.find((element) => element.textContent === keyContent);
+    keyElement.dispatchEvent(new Event('mouseup'));
+  }
+}
+
+function handleAltShift() {
+  let { locale } = keyboard.properties;
+  if (locale === 'rus') {
+    locale = 'en';
+  } else {
+    locale = 'rus';
+  }
+  window.localStorage.setItem('keyboardLocale', locale);
+  document.body.removeChild(keyboard.elements.main);
+  keyboard.init();
+}
+
+function handleBackspaceClick() {
+  const index = textarea.getCaret();
+  let value = textarea.getValue();
+  value = value.substring(0, index - 1) + value.substring(index, value.length);
+  textarea.setValue(value);
+}
+
+function handleEnterClick() {
+  const value = textarea.getValue();
+  textarea.setValue(`${value}\n`);
+}
+
+function handleSpaceClick() {
+  const value = textarea.getValue();
+  textarea.setValue(`${value} `);
+}
+
+function handleDeleteClick() {
+  const index = textarea.getCaret();
+  let value = textarea.getValue();
+  value = value.substring(0, index) + value.substring(index + 1, value.length);
+  textarea.setValue(value);
+}
+
+function handleKeyMouseDown(event) {
+  const { target } = event;
+  target.classList.add('keyboard__key_active');
+}
+
+function handleKeyMouseUp(event) {
+  const { target } = event;
+  target.classList.remove('keyboard__key_active');
+}
+
+function handleKeyMouseOut(event) {
+  const { target } = event;
+  target.classList.remove('keyboard__key_active');
+}
+
+function isCommonKey(key) {
+  return !(key.classList.contains('keyboard__key_medium')
+      || key.classList.contains('keyboard__key_wide')
+      || key.classList.contains('keyboard__key_extra-wide'));
+}
+
+function handleKeyClick(event) {
+  const { target } = event;
+  let value = textarea.getValue();
+  value += keyboard.getCapsLock()
+    ? target.textContent.toUpperCase() : target.textContent.toLowerCase();
+  textarea.setValue(value);
+}
+
+function handleCapslockClick() {
+  const capslock = keyboard.getCapsLock();
+  keyboard.setCapsLock(!capslock);
+
+  const keys = keyboard.getKeys();
+
+  keys.forEach((key) => {
+    if (isCommonKey(key)) {
+      key.textContent = keyboard.getCapsLock()
+        ? key.textContent.toUpperCase() : key.textContent.toLowerCase();
+    }
+  });
+}
+
+function handleShiftMouseDown() {
```

Comment:
these two you can combine into `toggle` thing and set as `keyboard.setCapslock(!capslock)`. Toggle function is a switch for boolean